Deepgram/Clients/Analyze/v1/Client.cs

```diff
@@ -0,0 +1,152 @@
+    {
+        VerifyOneCallBackSet(nameof(AnalyzeFileCallBack), callBack, analyzeSchema);
+        if (callBack != null)
+            analyzeSchema.CallBack = callBack;
+        var stringedOptions = QueryParameterUtil.GetParameters(analyzeSchema);
+        return await PostAsync<AsyncResponse>(
+            $"{UriSegments.READ}?{stringedOptions}",
+            RequestContentUtil.CreateStreamPayload(source), cancellationToken);
+    }
+
+    /// <summary>
+    /// Analyze a file by providing a url and a CallBack
+    /// </summary>
+    /// <param name="source">Url to the file that is to be analyzed <see cref="UrlSource"/></param>
+    /// <param name="callBack">CallBack url</param>    
+    /// <param name="analyzeSchema">Options for the transcription<see cref="AnalyzeSchema"></param>
+    /// <returns><see cref="AsyncResponse"/></returns>
+    public async Task<AsyncResponse> AnalyzeUrlCallBack(UrlSource source, string? callBack, AnalyzeSchema? analyzeSchema, CancellationToken cancellationToken = default)
+    {
+        VerifyOneCallBackSet(nameof(AnalyzeUrlCallBack), callBack, analyzeSchema);
+
+        if (callBack != null)
+            analyzeSchema.CallBack = callBack;
+        var stringedOptions = QueryParameterUtil.GetParameters(analyzeSchema);
+        return await PostAsync<AsyncResponse>(
+            $"{UriSegments.READ}?{stringedOptions}",
+            RequestContentUtil.CreatePayload(source), cancellationToken);
+    }
+    #endregion
+
+    #region CallbackChecks
+    private void VerifyNoCallBack(string method, AnalyzeSchema? analyzeSchema)
+    {
+        if (analyzeSchema != null && analyzeSchema.CallBack != null)
+            throw new ArgumentException($"CallBack cannot be provided as schema option to a synchronous transcription when calling {method}. Use {nameof(AnalyzeFileCallBack)} or {nameof(AnalyzeUrlCallBack)}");
+    }
+
+    private void VerifyOneCallBackSet(string callingMethod, string? callBack, AnalyzeSchema? analyzeSchema)
+    {
+
+        if (analyzeSchema.CallBack == null && callBack == null)
+        { //check if no CallBack set in either callBack parameter or AnalyzeSchema
+            var ex = new ArgumentException($"Either provide a CallBack url or set AnalyzeSchema.CallBack.  If no CallBack needed either {nameof(AnalyzeUrl)} or {nameof(AnalyzeFile)}");
+            Log.Exception(_logger, $"While calling {callingMethod} no callback set", ex);
+            throw ex;
+        }
+        else if (!string.IsNullOrEmpty(analyzeSchema.CallBack) && !string.IsNullOrEmpty(callBack))
+        {
+            //check that only one CallBack is set in either callBack parameter or AnalyzeSchema
+            var ex = new ArgumentException("CallBack should be set in either the CallBack parameter or AnalyzeSchema.CallBack not in both.");
+            Log.Exception(_logger, $"While calling {callingMethod}, callback set in both parameter and property", ex);
+            throw ex;
+        }
+    }
```

Comment:
Will address the TODOs for Logging in these issues since I haven't thought about how to address it yet: https://github.com/deepgram/deepgram-dotnet-sdk/issues/210 and https://github.com/deepgram/deepgram-dotnet-sdk/issues/219